abci/client/local_client.go

```diff
@@ -61,7 +61,9 @@ func (app *localClient) CheckTxAsync(ctx context.Context, req *types.CheckTxRequ
 }
 
 func (app *localClient) callback(req *types.Request, res *types.Response) *ReqRes {
-	app.Callback(req, res)
+	if app.Callback != nil {
```

Comment:
How come we weren't checking the functor for `nil` before?

Reply:
It was never the case that app.Callback was not set.